scipy/cluster/tests/test_vq.py

```diff
@@ -357,8 +350,8 @@ def test_kmeans_large_thres(self, xp):
         # Regression test for gh-1774
         x = xp.asarray([1, 2, 3, 4, 10], dtype=xp.float64)
         res = kmeans(x, xp.asarray(1), thresh=1e16)
-        assert_allclose(res[0], xp.asarray([4.]))
-        assert_allclose(res[1], 2.3999999999999999)
+        xp_assert_close(res[0], xp.asarray([4.]), check_dtype=False)
+        xp_assert_close(res[1], xp.asarray(2.3999999999999999)[()], check_dtype=False)
```

Comment:
`res[1]` is `float64`, but `xp.asarray(2.3999999999999999)[()]` is `float32` for `torch`.